### PR TITLE
Fix default rule drop flow metrics

### LINKED ISSUE #7143: sessiond: handle drop all rule stats properly for dropped metrics
### Your Environment

- **Version:** master 
- **Affected Component:** Access Gateway
- **Affected Subcomponent:** SessionD

### Describe the Issue
We currently ignore reported stats for `internal_default_drop_flow_rule`, which we need for the dropped tx/rx metrics. 
This feature broke when we migrated to flat rate reporting, tracking the breakage here.

### PR DESCRIPTION
**Pull request by koolzz**
Signed-off-by: Nick Yurchenko <koolzz@fb.com>

## Summary
Resolves #7143

## Test Plan

make test_session_manager   runs fine                        
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by karthiksubraveti — CHANGES_REQUESTED** [review]
Can you edit these queries, can you replace the method increase -> sum_over_time. Thanks.
https://github.com/magma/magma/blob/faba77bb55a87400dbc89733f6a2caabbafa0b3b/nms/app/packages/magmalte/app/views/subscriber/SubscriberChart.js#L77
```
  const queries = [
    {
      q: `sum(increase(ue_reported_usage{IMSI="${subscriberId}",direction="down"}[${step}]))`,
      color: colors.secondary.dodgerBlue,
      label: 'download',
    },
    {
      q: `sum(increase(ue_reported_usage{IMSI="${subscriberId}",direction="up"}[${step}]))`,
      color: colors.data.flamePea,
      label: 'upload',
    },
  ];
```

File: lte/gateway/c/session_manager/SessionState.cpp [hunk 1]
```diff
@@ -550,6 +549,10 @@ void SessionState::add_rule_usage(
       rule_id, rule_version, used_tx, used_rx, dropped_tx, dropped_rx,
       session_uc);
   if (!rule_delta) {
+    if (strcmp(rule_id.c_str(), DROP_ALL_RULE) == 0) {
```
Comment (themarwhal) [inline lte/gateway/c/session_manager/SessionState.cpp]: Should we compare first, and then skip `get_rule_delta` if it's a drop rule?

**Review by karthiksubraveti — COMMENTED** [review]
FYI This PR https://github.com/magma/magma/pull/7495#pullrequestreview-682150669 might conflict with your changes.

File: lte/gateway/c/session_manager/SessionState.cpp (hunk 1 above)
Comment (karthiksubraveti) [reply lte/gateway/c/session_manager/SessionState.cpp]: Also this being c++, you can simply add DROP_ALL_RULE as a c++ string and compare `rule_id == DROP_ALL_RULE`, this is much safer than using strcmp.

**Review by karthiksubraveti — APPROVED** [review]

**Review by uri200 — APPROVED** [review]

**Comment by themarwhal** [comment]
@koolzz + @karthiksubraveti 
do we only want the drop flow value to be gauge not counter?
I think this change may have modified `ue_reported_usage` metric to be gauge not counter as well. Just want to make sure that was intentional. Otherwise, I can revert the change for `ue_reported_usage`.

**Comment by themarwhal** [comment]
Or rather, if we want both to be gauges, is this correct?
https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/c/session_manager/SessionState.cpp?L591
`update_data_metrics(UE_USED_COUNTER_NAME, delta.tx, delta.rx);`


Resolving it here: https://github.com/magma/magma/pull/7495

**Pull request merged**